packages/cli/src/commands/index.ts

```diff
@@ -70,6 +70,10 @@ export function registerRepoCommand(program: Command) {
       'Set the success cache location, (default: node_modules/.cache/backstage-cli)',
     )
     .option('--fix', 'Attempt to automatically fix violations')
+    .option(
+      '--quiet',
+      'Report errors only. Rules marked as warnings do not cause non-zero exit.',
+    )
```

Comment:
Hmm, I don't think a `--quiet` flag should ever cause a change in behavior, only reduced logging. I'd actually think that this use-case is something that might already be covered by the `--max-warnings` flags for the individual `package lint`? That flag is picked up by `repo lint` too and lets you increase the max allowed warnings for each package.

Reply:
As I understood this, `--max-warnings` cannot be set for `repo lint` but must be set in package level. 

In our case, we are introducing some rules globally that will cause issues (warnings) with existing packages. I'd want to introduce this rule and start fixing these issues but it is not possible to do it in one go. And because of this, our CI will fail. 

I'd prefer not to set `--max-warnings` in package level, we do have quite a few of those.

Reply:
@Glenf the problem with adding this at the repo level is that it will break linting at the package level. Linting from the root might not lead to an error while running `lint` in a package will.

I'd really recommend setting the max warning per package as it also gives you a view into which packages have committed to the new lint rules and which still need some work.

Reply:
Out of curiosity, why are eslint rules marked as `warn` considered errors? 

And another question, if Backstage would use ESLint 9, would `--quiet` option supported by it be a better approach then?

Reply:
I did a little test on this and it seems that `--max-warnings` behaviour is different than what ESLint documents. Adding `--max-warnings=-1`, which should ignore all warnings in ESLint, does not have any effect and reports warnings as errors. This is also how the wrapper is implemented. 

Should I create a separate issue out of this?

Reply:
I can do a PR for those. I have a version ready from my testing but it needs a small cleanup.